src/main/kotlin/uk/gov/justice/digital/hmpps/healthandmedication/client/prisonersearch/PrisonerSearchClient.kt

```diff
@@ -21,4 +25,19 @@ class PrisonerSearchClient(@Qualifier("prisonerSearchWebClient") private val web
   } catch (e: Exception) {
     throw DownstreamServiceException("Get prisoner request failed", e)
   }
+
+  fun getPrisonersForPrison(prisonId: String, sort: String? = null): List<PrisonerDto>? {
+    try {
+      var query = "size=9999"
+      if (!sort.isNullOrEmpty()) query += "&sort=$sort"
```

Comment:
I think it's generally preferred to use the UriBuilder instead of concatenating strings, so something like:

```
      return webClient
        .get()
        .uri { uriBuilder ->
          uriBuilder.path("/prison/{prisonId}/prisoners")
            .queryParam("size", 9999)
            .queryParamIfPresent("sort", Optional.ofNullable(sort.takeUnless { it.isNullOrBlank() }))
            .build(prisonId)
        }
        .retrieve()
        .bodyToMono(PrisonerSearchResult::class.java)
        .block()?.content
```